Unit test failure

The following failure was found on the CI:

```
--- FAIL: TestGetTrafficTargetsWithDestinationInNamespace (0.81s)
    provider_test.go:164: 
        	Error Trace:	provider_test.go:164
        	Error:      	Not equal: 
        	            	expected: []*v1alpha1.TrafficTarget{(*v1alpha1.TrafficTarget)(0xc000133d40), (*v1alpha1.TrafficTarget)(0xc0004ded00)}
        	            	actual  : []*v1alpha1.TrafficTarget{(*v1alpha1.TrafficTarget)(0xc000283260), (*v1alpha1.TrafficTarget)(0xc0002830c0)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,30 +1,2 @@
        	            	 ([]*v1alpha1.TrafficTarget) (len=2) {
        	            	- (*v1alpha1.TrafficTarget)({
        	            	-  TypeMeta: (v1.TypeMeta) &TypeMeta{Kind:TrafficTarget,APIVersion:access.smi-spec.io/v1alpha1,},
        	            	-  ObjectMeta: (v1.ObjectMeta) &ObjectMeta{Name:api-service-metrics-2,GenerateName:,Namespace:default,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,ManagedFields:[],},
        	            	-  Status: (v1alpha1.Status) "",
        	            	-  Destination: (v1alpha1.IdentityBindingSubject) {
        	            	-   Kind: (string) (len=14) "ServiceAccount",
        	            	-   Name: (string) (len=11) "api-service",
        	            	-   Namespace: (string) (len=3) "foo",
        	            	-   Port: (string) ""
        	            	-  },
        	            	-  Sources: ([]v1alpha1.IdentityBindingSubject) (len=1) {
        	            	-   (v1alpha1.IdentityBindingSubject) {
        	            	-    Kind: (string) (len=14) "ServiceAccount",
        	            	-    Name: (string) (len=10) "prometheus",
        	            	-    Namespace: (string) (len=7) "default",
        	            	-    Port: (string) ""
        	            	-   }
        	            	-  },
        	            	-  Specs: ([]v1alpha1.TrafficTargetSpec) (len=1) {
        	            	-   (v1alpha1.TrafficTargetSpec) {
        	            	-    Kind: (string) (len=14) "HTTPRouteGroup",
        	            	-    Name: (string) (len=18) "api-service-routes",
        	            	-    Matches: ([]string) (len=1) {
        	            	-     (string) (len=7) "metrics"
        	            	-    }
        	            	-   }
        	            	-  }
        	            	- }),
        	            	  (*v1alpha1.TrafficTarget)({
        	            	@@ -62,2 +34,30 @@
        	            	   }
        	            	+ }),
        	            	+ (*v1alpha1.TrafficTarget)({
        	            	+  TypeMeta: (v1.TypeMeta) &TypeMeta{Kind:TrafficTarget,APIVersion:access.smi-spec.io/v1alpha1,},
        	            	+  ObjectMeta: (v1.ObjectMeta) &ObjectMeta{Name:api-service-metrics-2,GenerateName:,Namespace:default,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,ManagedFields:[],},
        	            	+  Status: (v1alpha1.Status) "",
        	            	+  Destination: (v1alpha1.IdentityBindingSubject) {
        	            	+   Kind: (string) (len=14) "ServiceAccount",
        	            	+   Name: (string) (len=11) "api-service",
        	            	+   Namespace: (string) (len=3) "foo",
        	            	+   Port: (string) ""
        	            	+  },
        	            	+  Sources: ([]v1alpha1.IdentityBindingSubject) (len=1) {
        	            	+   (v1alpha1.IdentityBindingSubject) {
        	            	+    Kind: (string) (len=14) "ServiceAccount",
        	            	+    Name: (string) (len=10) "prometheus",
        	            	+    Namespace: (string) (len=7) "default",
        	            	+    Port: (string) ""
        	            	+   }
        	            	+  },
        	            	+  Specs: ([]v1alpha1.TrafficTargetSpec) (len=1) {
        	            	+   (v1alpha1.TrafficTargetSpec) {
        	            	+    Kind: (string) (len=14) "HTTPRouteGroup",
        	            	+    Name: (string) (len=18) "api-service-routes",
        	            	+    Matches: ([]string) (len=1) {
        	            	+     (string) (len=7) "metrics"
        	            	+    }
        	            	+   }
        	            	+  }
        	            	  })
        	Test:       	TestGetTrafficTargetsWithDestinationInNamespace
```

It may be related to ordering of results.

Note that kubernetes does not guarantee any ordering with responses returned, and may return the same information in a different order at any time.

--- a/internal/providers/smi/provider_test.go
+++ b/internal/providers/smi/provider_test.go
@@ -161,7 +161,11 @@ func TestGetTrafficTargetsWithDestinationInNamespace(t *testing.T) {
 	assert.NoError(t, err)
 
 	actual := provider.getTrafficTargetsWithDestinationInNamespace("foo", allTrafficTargets)
-	assert.Equal(t, expected, actual)
+	assert.Equal(t, len(actual), len(expected))
+
+	for _, expectedValue := range expected {
+		assert.Contains(t, actual, expectedValue)
+	}
 }
 
 func TestBuildHTTPRouterFromTrafficTarget(t *testing.T) {

